source/memory_test.go

```diff
@@ -0,0 +1,317 @@
+// Copyright 2021 The Witness Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package source
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/testifysec/go-witness/attestation"
+	"github.com/testifysec/go-witness/dsse"
+	intoto "github.com/testifysec/go-witness/intoto"
+)
+
+func TestLoadEnvelope(t *testing.T) {
+	// Marshal the attestation.Collection into a JSON byte array
+	predicate, err := json.Marshal(attestation.Collection{})
+	if err != nil {
+		t.Fatalf("failed to marshal predicate, err = %v", err)
+	}
+
+	// Define the test cases
+	tests := []struct {
+		name                 string
+		reference            string
+		intotoStatment       intoto.Statement
+		mSource              *MemorySource
+		attCol               attestation.Collection
+		wantLoadEnvelopeErr  bool
+		wantPredicateErr     bool
+		wantMemorySourceErr  bool
+		wantRefrenceExistErr bool
+	}{
+		{
+			name:      "Valid intotoStatment",
+			reference: "ref",
+			intotoStatment: intoto.Statement{
+				Type:          "https://in-toto.io/Statement/v0.1",
+				Subject:       []intoto.Subject{{Name: "example", Digest: map[string]string{"sha256": "exampledigest"}}},
+				PredicateType: "https://slsa.dev/provenance/v0.2",
+				Predicate:     json.RawMessage(predicate),
+			},
+			attCol:  attestation.Collection{},
+			mSource: NewMemorySource(),
+		},
+		{
+			name:                "Empty Invalid intotoStatment",
```

Comment:
I understand that an empty in-toto statement has been provided, but maybe thinking "Invalid" in the test name might not be necessary. A bit of a nit though.